### ISSUE TITLE
Bun.Thread/Better workers

### ISSUE THREAD

**Issue by ghost** [oven-sh/bun#13091]
### What is the problem this feature would solve?

People have trouble working with workers to solve their multithreading problems and it's just too much to do as little as spin up a small worker that says hello world

### What is the feature you are proposing to solve the problem?

A new `Bun.Thread` class, it works like workers, but it's different, because unlike workers you don't need to create another file and tell it to use that file, you can just do `new Bun.Thread(function)` and that function will be ran in another thread, threads can be terminated using `.terminate`.

### What alternatives have you considered?

Instead of adding a new class, another solution would be to make the `Worker` class accept a function as well as a path or url.

**Comment by ezra-en** [comment]
if all you want is to be able to inline your workers, have you tried this? https://bun.sh/docs/api/workers#blob-urls

**Comment by codehz** [comment]
I don't think this is a good idea, js is not multithreaded safe by design, so you can't just pass a function to another thread, you can only do operations like toString, but this is obviously not a perfect solution because it can't capture the environment, and you can't use imports...

Some tc39 proposals are intended to solve this problem:
https://github.com/tc39/proposal-module-expressions (and they provide an example to show why this doesn't work.) which let you to write multiple module in a single file
https://github.com/tc39/proposal-module-declarations (like first one, but named)
https://github.com/tc39/proposal-defer-import-eval which let you to import a module but not eval it (so it can pass to worker, or send via postMessage

Of course, it is not a good idea to use these draft syntaxes at present (because they can change), but I think we can refer to the design ideas of these syntaxes to improve the current API.

(I think the module expression and module declaration is too hard to implemented in current syntax)

but emulate the defer import eval by using import assert is possible (although it is forbidden in the proposal), turn all imported symbols to a "reference", send those reference (with function parameter) to another worker and execute it in worker, so you no longer have to manually create a "Trampoline" function for every imported module...

**Comment by Jarred-Sumner** [comment]
We would love to do something like this (shared memory & objects, threads sharing a single global object)

But it'd take a small team of very experienced VM / Compiler engineers and lots of work in JavaScriptCore to pull it off. There are probably < 50 people in the world who have the relevant experience to build this today and make it work reliably.

**Comment by Autumn-one** [comment]
JavaScript has had this pain point throughout its long development. Unfortunately, it seems to require a lot of effort. The best scenario would naturally be the ability to create new threads like `Task.Run(function)` in C#. This is a beautiful aspiration. Currently, however, the performance of workers is truly terrible, and the syntax can simply be described as outdated.
